control-plane-operator/controllers/hostedcontrolplane/ingress/router.go

```diff
@@ -40,86 +41,141 @@ func HCPRouterConfig(hcp *hyperv1.HostedControlPlane, setDefaultSecurityContext
+					corev1.ResourceCPU:    resource.MustParse("50m"),
 				},
 			},
 		},
 	}
-	cfg.LivenessProbes = config.LivenessProbes{
-		hcpRouterContainerMain().Name: corev1.Probe{
-			FailureThreshold: 3,
-			ProbeHandler: corev1.ProbeHandler{
-				HTTPGet: &corev1.HTTPGetAction{
-					Path:   "/healthz",
-					Port:   intstr.FromInt(metricsPort),
-					Scheme: corev1.URISchemeHTTP,
-				},
-			},
-			PeriodSeconds:    10,
-			SuccessThreshold: 1,
-			TimeoutSeconds:   1,
-		},
-	}
-	cfg.ReadinessProbes = config.ReadinessProbes{
-		hcpRouterContainerMain().Name: corev1.Probe{
-			FailureThreshold: 3,
-			ProbeHandler: corev1.ProbeHandler{
-				HTTPGet: &corev1.HTTPGetAction{
-					Path:   "/healthz/ready",
-					Port:   intstr.FromInt(metricsPort),
-					Scheme: corev1.URISchemeHTTP,
-				},
-			},
-			PeriodSeconds:    10,
-			SuccessThreshold: 1,
-			TimeoutSeconds:   1,
-		},
-	}
+
 	cfg.Scheduling.PriorityClass = config.APICriticalPriorityClass
-	cfg.SetDefaults(hcp, hcpRouterLabels(), nil)
+	cfg.SetRequestServingDefaults(hcp, hcpRouterLabels(), nil)
 	cfg.SetRestartAnnotation(hcp.ObjectMeta)
 	cfg.SetDefaultSecurityContext = setDefaultSecurityContext
 	return cfg
 }
 
 const PrivateRouterImage = "haproxy-router"
 
-const (
-	routerTemplateConfigMapKey = "haproxy-config.template"
-	routerTemplateVolumeName   = "happroxy-config"
-)
+//go:embed router_config.template
+var routerConfigTemplateStr string
+var routerConfigTemplate *template.Template
+
+func init() {
+	var err error
+	routerConfigTemplate, err = template.New("router-config").Parse(routerConfigTemplateStr)
+	if err != nil {
+		panic(err.Error())
+	}
+}
+
+type byRouteName []routev1.Route
+
+func (r byRouteName) Len() int           { return len(r) }
+func (r byRouteName) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
+func (r byRouteName) Less(i, j int) bool { return r[i].Name < r[j].Name }
+
+func generateRouterConfig(namespace string, kubeAPIPort int32, routeList *routev1.RouteList, nameServerIP string) (string, error) {
+	type backendDesc struct {
+		Name               string
+		HostName           string
+		DestinationService string
+		DestinationPort    int32
+	}
+	type templateParams struct {
+		HasKubeAPI   bool
+		Namespace    string
+		KubeAPIPort  int32
+		Backends     []backendDesc
+		NameServerIP string
+	}
+	p := templateParams{
+		Namespace:    namespace,
+		NameServerIP: nameServerIP,
+	}
+	sort.Sort(byRouteName(routeList.Items))
+	for _, route := range routeList.Items {
+		if _, hasHCPLabel := route.Labels[util.HCPRouteLabel]; !hasHCPLabel {
+			// If the hypershift.openshift.io/hosted-control-plane label is not present,
+			// then it means the route should be fulfilled by the management cluster's router.
+			continue
+		}
+		switch route.Name {
+		case manifests.KubeAPIServerInternalRoute("").Name,
+			manifests.KubeAPIServerExternalPublicRoute("").Name,
+			manifests.KubeAPIServerExternalPrivateRoute("").Name:
+			p.HasKubeAPI = true
+			continue
```

Comment:
who do we need `continue` here?

Reply:
just curious, otherwise lgtm

Reply:
we don't need it :) 
I had that there before that section was inside the case stmt but forgot to remove. It doesn't hurt, but I will remove it in a follow up